src/core/meshcop/tcat_agent.cpp

```diff
@@ -506,9 +514,11 @@ Error TcatAgent::HandleSingleTlv(const Message &aIncomingMessage, Message &aOutg
 
 Error TcatAgent::HandleSetActiveOperationalDataset(const Message &aIncomingMessage, uint16_t aOffset, uint16_t aLength)
 {
-    Dataset     dataset;
-    OffsetRange offsetRange;
-    Error       error;
+    Dataset       dataset;
+    OffsetRange   offsetRange;
+    Error         error;
+    unsigned char buf[1024];
```

Comment:
The value 1024 is used multiple times. I suggest introducing a const to ensure consistency.

In class Flash  kMaxDataSize is 512.  Isn't that the maximum expected here as well?

Reply:
See earlier comment - cert can be up to 1024B as currently tentatively defined in TCAT (BHC-... about sizes)

Reply:
(The JIRA with 1K size proposal is https://threadgroup.atlassian.net/browse/BHC-744 . Also for CCM certs, the Conformance Doc specifies 1K currently.)